app/furniture/marketplace/cart.rb

```diff
@@ -1,30 +1,27 @@
 # frozen_string_literal: true
 
 class Marketplace
-  class Cart < ApplicationRecord
+  class Cart < Record
     self.table_name = "marketplace_carts"
-    include WithinLocation
     self.location_parent = :marketplace
 
+    default_scope { where(status: :pre_checkout) }
```

Comment:
This ensures whenever we try and find a Cart it doesn't include anything that has been checked out.